src/cpp/fastdds/xtypes/type_conversion/dyn_type_tree.ipp

```diff
@@ -0,0 +1,1116 @@
+        return ret;
+    }
+
+    std::string key_str;
+    const auto key_type = type_descriptor->key_element_type();
+    ret = type_kind_to_str(key_type, key_str);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    std::string value_str;
+    const auto value_type = type_descriptor->element_type();
+    ret = type_kind_to_str(value_type, value_str);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    std::stringstream ss;
+
+    ss << "map<" << key_str << ", " << value_str << ">";
+
+    map_str = ss.str();
+
+    return ret;
+}
+
+ReturnCode_t sequence_kind_to_str(
+        const DynamicType::_ref_type& dyn_type,
+        std::string& sequence_str) noexcept
+{
+    if (dyn_type->get_kind() != TK_SEQUENCE)
+    {
+        EPROSIMA_LOG_ERROR(DYN_TYPES, "Type is not a sequence.");
+        return RETCODE_BAD_PARAMETER;
+    }
+
+    ReturnCode_t ret = RETCODE_OK;
+
+    DynamicType::_ref_type internal_type;
+    ret = container_internal_type(dyn_type, internal_type);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    ret = type_kind_to_str(internal_type, sequence_str);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    sequence_str = "sequence<" + sequence_str;
+
+    BoundSeq bounds;
+    ret = container_size(dyn_type, bounds);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    for (const auto& bound : bounds)
+    {
+        const auto UNBOUNDED = static_cast<std::uint32_t>(LENGTH_UNLIMITED);
+
+        if (bound != UNBOUNDED)
+        {
+            sequence_str += ", " + std::to_string(bound);
+        }
+    }
+
+    sequence_str += ">";
+
+    return ret;
+}
+
+ReturnCode_t get_members_sorted(
+        const DynamicType::_ref_type& dyn_type,
+        std::vector<std::pair<std::string, MemberDescriptor::_ref_type>>& result) noexcept
+{
+    ReturnCode_t ret = RETCODE_OK;
+
+    // Skip the inherited members
+    TypeDescriptor::_ref_type type_descriptor {traits<TypeDescriptor>::make_shared()};
+    ret = dyn_type->get_descriptor(type_descriptor);
+
+    if (ret != RETCODE_OK)
+    {
+        return ret;
+    }
+
+    const auto base_type = type_descriptor->base_type();
+
+    std::uint32_t first_member = 0;
+
+    if (base_type != nullptr)
+    {
+        // If the struct has a base type, the first member is the base type
+        first_member = base_type->get_member_count();
+    }
+
+    // Collect its members
+    for (std::uint32_t index = first_member; index < dyn_type->get_member_count(); index++)
+    {
+        traits<DynamicTypeMember>::ref_type member;
+        ret = dyn_type->get_member_by_index(member, index);
+
+        if (ret != RETCODE_OK)
+        {
+            return ret;
+        }
+
+        MemberDescriptor::_ref_type member_descriptor {traits<MemberDescriptor>::make_shared()};
+        ret = member->get_descriptor(member_descriptor);
+
+        if (ret != RETCODE_OK)
+        {
+            return ret;
+        }
```

Comment:
You could benefit from the Impl class to simplify the code a bit, as well as for efficiency.